chasm/lib/activity/activity.go

```diff
@@ -679,3 +696,103 @@ func (a *Activity) buildPollActivityExecutionResponse(
 		FrontendResponse: response,
 	}, nil
 }
+
+// recordOnAttemptedMetrics records metrics for attempted activities, including retries and originating from any
+// terminal state transitions.
```

Comment:
I don't quite follow the "and originating from any terminal state transitions" bit.

For the first bit, would "records metrics for an activity attempt" be clearer?